golos.publication/objects.hpp

```diff
@@ -199,7 +210,9 @@ namespace tables {
 using namespace eosio;
 
 using id_index = indexed_by<N(id), const_mem_fun<structures::message, uint64_t, &structures::message::primary_key>>;
-using message_table = multi_index<N(message), structures::message, id_index>;
+using permlink_index = indexed_by<N(bypermlink), const_mem_fun<structures::message, std::string, &structures::message::secondary_key>>;
```

Comment:
We have rich functional to work with secondary indexes. See https://github.com/GolosChain/golos.contracts/pull/415/files as example